src/openssl_wrap.cpp

```diff
@@ -1323,5 +1323,29 @@ std::vector<uint8_t> _EC_KEY_key2buf(const EVP_PKEY* evp, point_conversion_form_
     return result;
 }
 
+
+std::vector<uint8_t> _ECDH_compute_key(const EVP_PKEY *pub_key, const EVP_PKEY *ecdh,
+                                       void *(*KDF) (const void *in, size_t inlen, void *out, size_t *outlen))
+{
+    EVP_PKEY *pub_key_ = const_cast<EVP_PKEY*>(pub_key);
+    EVP_PKEY *ecdh_ = const_cast<EVP_PKEY*>(ecdh);
+    EC_KEY *ec_key = OpensslCallPtr::callChecked(lib::OpenSSLLib::SSL_EVP_PKEY_get1_EC_KEY, ecdh_);
```

Comment:
Doesn't `EVP_PKEY_get1_EC_KEY` increase the reference count of the key and thus requires an associated free? `EC_KEY` does not seem to be a smart pointer that does this automatically (but maybe it should be)?

Did you run this under valgrind or with address sanitizer?

Reply:
The more I think about this the more I think our SSL_EVP_PKEY_get1_EC_KEY wrapper should return a smart pointer.

Reply:
Done